limbus/core/component.py

```diff
@@ -154,9 +164,9 @@ def stopping_iteration(self) -> int:
         return self._stopping_iteration
 
     @property
-    def state(self) -> ComponentState:
-        """Get the current state of the component."""
-        return self._state.state
+    def state(self) -> Tuple[ComponentState, Optional[str]]:
```

Comment:
```suggestion
    def state(self) -> tuple[ComponentState, str | None]:
```
from future annotations